### ISSUE TITLE
Slicing

### ISSUE THREAD

**Issue by cortner** [JuliaArrays/StaticArrays.jl#149]
I would expect that slices of static arrays are again static arrays, but this is not so:
```
julia> a
3×4×4 StaticArrays.SArray{(3,4,4),Float64,3,48}:
[:, :, 1] =
 0.0  0.0  0.0  0.0
 0.0  0.0  0.0  0.0
 0.0  0.0  0.0  0.0

[:, :, 2] =
 0.0  0.0  0.0  0.0
 0.0  0.0  0.0  0.0
 0.0  0.0  0.0  0.0

[:, :, 3] =
 0.0  0.0  0.0  0.0
 0.0  0.0  0.0  0.0
 0.0  0.0  0.0  0.0

[:, :, 4] =
 0.0  0.0  0.0  0.0
 0.0  0.0  0.0  0.0
 0.0  0.0  0.0  0.0

julia> a[:, 2, 3]
3-element Array{Float64,1}:
 0.0
 0.0
 0.0
```
For me it would be useful to have this feature, but maybe it is intended as it is now?

(I fix this for myself for special cases as needed, but I am unsure about a clean generic implementation, especially for `SArray`, otherwise I'd offer to make a PR myself.)

**Comment by andyferris** [comment]
Is this Julia v0.5 or v0.6?

**Comment by cortner** [comment]
I am still on v0.5. Base on our last conversation I can immagine that this might be easier on v0.6. But as I said I can always do ad-hoc fixes for now so I wouldn't worry about v0.5 on my behalf. (+ I hope to migrate to v0.6 within the next few months.)

**Comment by andyferris** [comment]
Ok. This is *should* already be fixed in v0.6 code, which supports the full multidimensional APL slicing rules. 

Some improvements like this would be backportable, but I'm too lazy. PRs welcome, of course ;)

**Comment by cortner** [comment]
then I won't stress about it - sorry I didn't think to check v0.6.
